avro4s-core/src/main/scala/com/cognitops/avro4s/CustomDecoder.scala

```diff
@@ -0,0 +1,19 @@
+package com.cognitops.avro4s
+
+import java.time.OffsetDateTime
+import java.time.format.DateTimeFormatter
+
+import com.sksamuel.avro4s.{Decoder, FieldMapper}
+import org.apache.avro.Schema
+
+/**
+ * Custom Cognitops Decoders
+ */
+object CustomDecoder {
+
+  implicit object OffsetDateTimeDecoder extends Decoder[OffsetDateTime] {
+    override def decode(value: Any, schema: Schema, fieldMapper: FieldMapper) =
+      OffsetDateTime.parse(value.toString, DateTimeFormatter.ISO_OFFSET_DATE_TIME)
```

Comment:
This can throw an exception. Is that what Avro expects to happen if the value can't be decoded correctly? I'm guessing yes since the method returns an OffsetDateTime and not an Either. Just making sure.

Reply:
Yes. The other Decoders in the library throw exceptions.

Reply:
Sounds good